src/acvp_build_register.c

```diff
@@ -1666,6 +1679,83 @@ static ACVP_RESULT acvp_build_kdf108_register_cap(JSON_Object *cap_obj, ACVP_CAP
     return ACVP_SUCCESS;
 }
 
+static ACVP_RESULT acvp_build_kdf108_kmac_register_cap(JSON_Object *cap_obj, ACVP_CAPS_LIST *cap_entry) {
+    ACVP_RESULT result;
+    JSON_Array *tmp_arr = NULL;
+    JSON_Value *tmp_val = NULL;
+    JSON_Object *tmp_obj = NULL;
+    ACVP_NAME_LIST *nl_obj;
+    ACVP_KDF108_MODE_PARAMS *params;
+
+    /* is KMAC enabled? */
+    params = &cap_entry->cap.kdf108_cap->kmac_mode;
+    if (!params->kdf_mode) {
+        return ACVP_SUCCESS;
+    }
+
+    json_object_set_string(cap_obj, "algorithm", "KDF");
+    json_object_set_string(cap_obj, "mode", "KMAC");
+
+    /* Revision is diff from base kdf108.  Should it be new algorithm? */
+    json_object_set_string(cap_obj, "revision", ACVP_REV_STR_SP800_108R1);
+
+    result = acvp_lookup_prereqVals(cap_obj, cap_entry);
+    if (result != ACVP_SUCCESS) { return result; }
+
+    params = &cap_entry->cap.kdf108_cap->kmac_mode;
+
+    /* mac mode list */
+    json_object_set_value(cap_obj, "macMode", json_value_init_array());
+    tmp_arr = json_object_get_array(cap_obj, "macMode");
+    nl_obj = params->mac_mode;
+    while (nl_obj) {
+        json_array_append_string(tmp_arr, nl_obj->name);
+        nl_obj = nl_obj->next;
+    }
+
+    /* key derivation key length list */
+    json_object_set_value(cap_obj, "keyDerivationKeyLength", json_value_init_array());
+    tmp_arr = json_object_get_array(cap_obj, "keyDerivationKeyLength");
+    tmp_val = json_value_init_object();
+    tmp_obj = json_value_get_object(tmp_val);
+    json_object_set_number(tmp_obj, "min", params->derivation_keylens.min);
+    json_object_set_number(tmp_obj, "max", params->derivation_keylens.max);
+    json_object_set_number(tmp_obj, "increment", params->derivation_keylens.increment);
```

Comment:
We also want to append all the "values" in the list from ACVP_JSON_DOMAIN_OBJ to the array to stay in line with the domain definition in the spec (this hasn't been retroactively added to older algorithms yet, so it isn't very clear, its a very new change). Thanks!

Reply:
I can't find reference to the change in the ACVP spec, nor can I find an example in code (all the *_domain_set() functions don't ref the "values" field in the domain object).  Can you help find a ref to understand the requested change?